datafusion/core/tests/sqllogictests/src/output.rs

```diff
@@ -0,0 +1,26 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+use sqllogictest::{DBOutput, DefaultColumnType};
+
+pub type DFColumnType = DefaultColumnType;
```

Comment:
DBOutput is now generic to support a custom column type description.

cc @melgenek  as I think you were discussing this on a previous PR

Reply:
Yes, sqllogictest was updated.

The update in this pr doesn't:
- validate types by default. By default, any types are successful as long as it can be parsed. Validation can be enabled with `runner.with_column_validator(strict_column_validator);`
- `--complete` doesn't fill in the expected types. I need to create another pr in `sqllogictest-rs`.

Anyway, I was going to work on https://github.com/apache/arrow-datafusion/issues/4499 this weekend. If you give me a couple of days, I'll create a pr with type validation.